### PR TITLE
[PRDP-54] feat: updated storage policies

### PR DESCRIPTION
**Pull request by alessio-cialini**
### List of changes

- Defining last access time property to enabled for storage module
- Changed policy from modification to last access time based
- Introducing deletion and archival variables and policies based on last access time (deletion after 10 years)

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/domains/receipts-common/99_variables.tf [hunk 1]
```diff
@@ -161,13 +161,19 @@ variable "receipts_datastore_fn_sa_delete_retention_days" {
...
-  default     = 120
+  default     = 183
 }
 
-variable "receipts_datastore_fn_sa_tier_to_cool_after_modification" {
+variable "receipts_tier_to_archive_after_days_since_last_access_time_greater_than" {
   type        = number
-  description = "Number of days since modification to blob before moving to cool tier"
-  default     = 120
+  description = "Number of days since last access to blob before moving to cool tier"
```
Comment (pasqualespica) [inline src/domains/receipts-common/99_variables.tf:169]: same description as above ? its right ?

File: src/domains/receipts-common/03_storage.tf [hunk 2]
```diff
@@ -84,6 +84,8 @@ module "receipts_datastore_fn_sa" {
   blob_delete_retention_days = var.receipts_datastore_fn_sa_delete_retention_days
   tags                       = var.tags
 
+  blob_last_access_time_enabled = true
```
Comment (pasqualespica) [inline src/domains/receipts-common/03_storage.tf:87]: to try it , we need to change this
`"git::https://github.com/pagopa/terraform-azurerm-v3.git//storage_account?ref=v6.7.0"`
in
`"git::https://github.com/pagopa/terraform-azurerm-v3.git//storage_account?ref=PRDP-54-feat-introduce_last_access_time_in_storage"`

**Review by pasqualespica — CHANGES_REQUESTED** [review]

File: src/domains/receipts-common/03_storage.tf (hunk 2 above)
Comment (alessio-cialini) [reply src/domains/receipts-common/03_storage.tf:87]: Done

**Review by pasqualespica — APPROVED** [review]
🚀

**Pull request merged**